### DISCUSSION TITLE
Simple performance boost when using pagination with pagination_total disabled

### DISCUSSION THREAD

**Discussion by sdeming** [activeadmin/activeadmin#8347] [Ideas]
In the ActiveAdmin::Views::PaginatedCollection#build_pagination method there is code to avoid making a query to get the count when you disable `pagination_total`. 

I suggest making a small improvement to send `reorder(nil)` to the collection for the one query that is made. Since ordering has no impact on the size of the results it should have no material impact but the query will run significantly faster.

Suggested change looks like this:

```ruby
module ActiveAdmin
  module Views
    class PaginatedCollection
      ...
      def build_pagination
        options = { views_prefix: :active_admin, outer_window: 1, window: 2 }
        options[:params] = @params if @params
        options[:param_name] = @param_name if @param_name

        if !@display_total
          # The #paginate method in kaminari will query the resource with a
          # count(*) to determine how many pages there should be unless
          # you pass in the :total_pages option. We issue a query to determine
          # if there is another page or not, but the limit/offset make this
          # query fast.
          offset = @collection.reorder(nil).offset(@collection.current_page * @collection.limit_value).limit(1).count
          options[:total_pages] = @collection.current_page + offset
          options[:right] = 0
        end

        text_node paginate @collection, **options
      end
      ...
    end
  end
end
```

For my specific use case it changes the query time from about 3s to < 100ms and it comes for free.

**Comment by mgrunberg** [comment]
@sdeming there is an open PR with your idea https://github.com/activeadmin/activeadmin/pull/6911

**Reply by sdeming** [reply]
This is great! Thank you for doing this @mgrunberg.

**Reply by mgrunberg** [reply]
You are welcome. The fix is included in v3.2.5.
